core/src/avm1/globals/array.rs

```diff
@@ -789,11 +788,10 @@ fn sort_compare_fields<'a, 'gc: 'a>(
     field_names: Vec<String>,
     mut compare_fns: Vec<CompareFn<'a, 'gc>>,
 ) -> impl 'a + FnMut(&mut Activation<'_, 'gc, '_>, &Value<'gc>, &Value<'gc>) -> Ordering {
-    use crate::avm1::object::value_object::ValueObject;
     move |activation, a, b| {
         for (field_name, compare_fn) in field_names.iter().zip(compare_fns.iter_mut()) {
-            let a_object = ValueObject::boxed(activation, *a);
-            let b_object = ValueObject::boxed(activation, *b);
+            let a_object = (*a).coerce_to_object(activation);
+            let b_object = (*b).coerce_to_object(activation);
```

Comment:
I think this can just be `a.coerce_to_object(activation)` too, because of auto-deref rules.